### PR TITLE
Update TypeScript to 2.6.2

### LINKED ISSUE #3265: Upgrade typescript-eslint-parser to a final version instead of a specific commit
Hi,

I've noticed that the dependency to typescript-eslint-parser is fixed to a certain commit:

`git://github.com/eslint/typescript-eslint-parser.git#9c71a627da36e97da52ed2731d58509c952b67ae`

But it seems that the version 8.0.1 of typescript-eslint-parser was released some days later, so why not upgrade? ;)

Version 9.0.0 of that parser has major changes due to TS 2.6 so maybe just 8.0.1 should be fine for the moment...

### PR DESCRIPTION
**Pull request by duailibe**
Bumping TS to 2.6.2 for JSX fragments support (#3237)

---

Closes #3265 (added by @ikatyang)

### CODE REVIEW

**Comment by duailibe** [comment]
1) Is `TSFirstContextualKeyword` being used instead of `TSAbstractKeyword` because it's a insiders version? cc @azz @ikatyang 

2) There's a comment being moved because of a change in the end pos of TypeInstantiation inside TypeReference. I'm opening a PR in https://github.com/eslint/typescript-eslint-parser

File: package.json [hunk 1]
```diff
@@ -46,8 +46,8 @@
     "remark-parse": "4.0.0",
     "semver": "5.4.1",
     "string-width": "2.1.1",
-    "typescript": "2.5.3",
-    "typescript-eslint-parser": "git://github.com/eslint/typescript-eslint-parser.git#9c71a627da36e97da52ed2731d58509c952b67ae",
+    "typescript": "2.6.2-insiders.20171120",
```
Comment (ikatyang) [inline package.json]: Isn't it a [TS 2.7 feature](https://github.com/Microsoft/TypeScript/wiki/Roadmap#27-january-2018)?

Comment (duailibe) [reply package.json]: They've backported to 2.6.2 (https://github.com/Microsoft/TypeScript/pull/19249#issuecomment-340896466, https://github.com/Microsoft/TypeScript/pull/20002)

**Comment by duailibe** [comment]
Thanks @ikatyang. But do you know if that reference gets stripped in the public version (since it's marked with `@internal`)? Meaning: should I just wait for the public 2.6.2 release, or should I add the AST node type to prettier?

I've opened a PR https://github.com/eslint/typescript-eslint-parser/pull/405

**Comment by ikatyang** [comment]
IIRC, there's no difference between their `@latest`/`@insider`/`@next` build process.

I'm not sure if we should add it here or in typescript-eslint-parser, cc @azz.

**Comment by duailibe** [comment]
Oh, I've just noticed those two lines are new.. I thought they've always been there

File: package.json (hunk 1 above)
Comment (ikatyang) [reply package.json]: image

**Comment by duailibe** [comment]
Here you can see it failing on "unknown type: TSJsxFragment"

[deploy preview](https://deploy-preview-3331--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeAfAHSgAlQEwEsA3deADxlQHojTs86y5Kansb0QAaECABxiFoAZ2SgAhiJjIAZhIA2IuLwBGAJwlgA1nBgBlflsJQA5shjqAripD4IYOYuW8Ty9TAAKm0wFsJTkq2AFYi5ABCmjp6+hK+cAAyJnCBLnxWMPwZAEyptkbq7sggMACe-HAiYOqEgjwg-DWwAOqE+DAAFsgAHAAMvI0Qys2a-BbWtupwAI5WhFPeEn4BSPJBvMq+hOM2GyamCnAAilYQ8Hm8MBKqre1dSNmXmoQK+wDCEL7+xVDQKbxWZQAFWuYlWzjgAF9IUA)

So glad we did this :-)

**Comment by duailibe** [comment]
@azz Do you have any opinions about https://github.com/prettier/prettier/pull/3331#issuecomment-347184335, if we should fix here or on the typescript-eslint-parser?

**Comment by JamesHenry** [comment]
Thanks for driving this @duailibe 👍 

Feel free to always @ me in these PRs and we can get things updated immediately

**Comment by duailibe** [comment]
Thank you, @JamesHenry!

Will do, thanks :-)

**Comment by duailibe** [comment]
Travis is finishing up, 3/4 is already green, after if finishes, I'm gonna merge to unlock the JSX fragments PR.

**Pull request merged**